src/screens/PinFlow/CurrentPinScreen.tsx

```diff
@@ -27,29 +28,27 @@ interface Props {
   };
   setTimeCounter: (timestamp: number) => SetTimeCounterAction;
   setFailedAttempts: (attempt: number) => SetFailedAttemptsAction;
+  setFailedAttemptStep: (failedAttempt: number) => SetFailedAttemptStepAction;
   timeCounter: TimeCounterState;
 }
 
 interface State {
   pin: string;
   error: string;
-  failedTimes: number;
   isCount: boolean;
 }
 
 class CurrentPinScreen extends PureComponent<Props, State> {
   state = {
     pin: '',
     error: '',
-    failedTimes: 0,
     isCount: true,
   };
 
-  handleFailedAttempt = (increasedFailedTimes: number) => {
+  handleFailedAttempt = (increasedFailedAttemptStep: number) => {
     const { attempt } = this.props.timeCounter;
-    const isFinalAttempt = increasedFailedTimes > 2;
-    const finalAttempt = 3;
-    const firstAttempt = 0;
+    const { setFailedAttempts, setFailedAttemptStep, setTimeCounter } = this.props;
+    const isFinalAttempt = increasedFailedAttemptStep > 2;
```

Comment:
` const isFinalAttempt = increasedFailedAttemptStep === finalAttempt ` would tell more info what is done here